src/main/sarl/fr/utbm/info/ia51/crowdproject/motion/MotionAlgorithms.sarl

```
@@ -43,9 +47,23 @@ class KinematicWanderAlgorithm implements WanderAlgorithm {
  */
 class SteeringWanderAlgorithm implements WanderAlgorithm {
 
-	override run(emitter : UUID, currentPosition : Vector2) : MotionQuery {
-		val query = new MotionQuery(emitter, MotionBehavior::STEERING)
-		// TODO
-		return query
+	override run(emitter : UUID, currentPosition : Vector2, orientation : double, velocity : Vector2) : MotionQuery {
```

Do you really need the velocity? Can't you compute it instead?

---

We need it at each step in this algorithm so i think it's better to avoid a recalculation : we already have it

---

Yay, except passing informations the agent about is a bad idea. A simple recalculation with cos/sin is a better idea. It's a small calculus for a better encapsulation.

---

uh.... I need more information to pass if i want the velocity : the speed is not supposed to be constant for example so i need it to calculate the velocity. If I remove the velocity, I'll need other things to pass...

---

I don't understand... Why do you need the velocity again?

---

look at the steering wander behavior l.52 and 56

---

You don't need that. A simple basis vector rotated along the `rotation` parameter is enough. You just have to define what your basis vector will be.

---

Ok, i change that